pages/pipelines/migrate_from_jenkins.md

```diff
@@ -0,0 +1,204 @@
+
+You'll continue to adjust the agent configuration as you monitor performance to optimize build times and resource usage for your needs.
+
+See the [Installation](/docs/agent/v3/installation/) guides when you're ready to install an agent and follow the instructions for your infrastructure type.
+
+## Translate pipeline definitions
+
+A pipeline is a container for modeling and defining workflows. While that's true for both Buildkite and Jenkins, they look quite different. Both can read a configuration file checked into a repository to define the workflow. In Jenkins, the `Jenkinsfile`. In Buildkite, the `pipeline.yml`. Where the `Jenkinsfile` uses a Groovy-based syntax and strong hierarchy, `pipelines.yml` uses YAML and a flat structure for better readability.
+
+### Audit your pipelines
+
+Before you start moving pipelines, we recommend taking inventory of your existing pipelines, plugins, and integrations. Determine which parts of your Jenkins setup are essential and which can be replaced or removed. This will help you decide what needs to be migrated to Buildkite.
+
+### Translate a pipeline
+
+Since the configuration files are quite different, creating an automated tool to translate between them is difficult. Instead, we recommend assessing the goal of a pipeline and investing the time to see how to achieve the same thing the Buildkite way. This results in clearer pipelines with better performance.
+
+Some Buildkite features you might want to use include [dynamic pipelines](/docs/pipelines/defining-steps#dynamic-pipelines), [lifecycle hooks](/docs/agent/v3/hooks), [conditionals](/docs/pipelines/conditionals), [artifacts](/docs/pipelines/artifacts), and [build matrices](/docs/pipelines/build-matrix).
+
+A simple pipeline in Buildkite might look like the following:
+
+```yaml
+steps:
+- label: "Build"
+  command: "build.sh"
+  key: "build"
+
+- label: "Test"
+  command: "test.sh"
+  key: "test"
+  depends_on: "build"
+
+- label: "Deploy"
+  command: "deploy.sh"
+  depends_on: "build"
+```
+
+To translate a pipeline:
+
+1. Identify the goal of the pipeline.
+1. Look for an [example pipeline](/docs/pipelines/example-pipelines) closest to that goal.
+1. Follow [Defining steps](/docs/pipelines/defining-steps) and surrounding documentation to learn how to customize the pipeline definition to meet your needs, including:
+   * Targeting a specific agent or queue.
+   * Replacing any Jenkins plugins and integrations with native integrations, existing Buildkite plugins, custom plugins, or custom scripts.
+1. Migrate any environment variables, secrets, or credentials used in the pipeline. Buildkite allows you to manage environment variables and secrets on different levels, such as organization, pipeline, and step levels. Securely store your sensitive data using Buildkite's secret management or a third-party secret management tool.
```

Comment:
This needs a re-write, Buildkite doesn't have secret management built in.

Reply:
👍🏻 Addressed